platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java

```diff
@@ -83,56 +230,290 @@ public PlatformState copy() {
     }
 
     /**
-     * Set the object containing miscellaneous platform information.
+     * Get the round when this state was generated.
      *
-     * @param round
-     * 		round data
+     * @return a round number
      */
-    public void setPlatformData(final PlatformData round) {
-        setChild(ChildIndices.PLATFORM_DATA, round);
+    public long getRound() {
+        return round;
     }
 
     /**
-     * Get the previous address book.
+     * Set the round when this state was generated.
+     *
+     * @param round a round number
      */
-    public AddressBook getPreviousAddressBook() {
-        return getChild(ChildIndices.PREVIOUS_ADDRESS_BOOK);
+    public void setRound(final long round) {
+        this.round = round;
     }
 
     /**
-     * Set the previous address book.
+     * Get the running hash of all events that have been applied to this state since the beginning of time.
+     *
+     * @return a running hash of events
+     */
+    public Hash getHashEventsCons() {
+        return hashEventsCons;
+    }
+
+    /**
+     * Set the running hash of all events that have been applied to this state since the beginning of time.
+     *
+     * @param hashEventsCons a running hash of events
+     */
+    public void setHashEventsCons(final Hash hashEventsCons) {
+        this.hashEventsCons = hashEventsCons;
+    }
+
+    /**
+     * Get the consensus timestamp for this state, defined as the timestamp of the first transaction that was applied in
+     * the round that created the state.
+     *
+     * @return a consensus timestamp
+     */
+    public Instant getConsensusTimestamp() {
+        return consensusTimestamp;
+    }
+
+    /**
+     * Set the consensus timestamp for this state, defined as the timestamp of the first transaction that was applied in
+     * the round that created the state.
+     *
+     * @param consensusTimestamp a consensus timestamp
+     */
+    public void setConsensusTimestamp(final Instant consensusTimestamp) {
+        this.consensusTimestamp = consensusTimestamp;
+    }
+
+    // TODO remove methods that are not simple getters and setters
+
+    /**
+     * Get the minimum event generation for each node within this state.
+     *
+     * @return minimum generation info list, or null if this is a genesis state
+     */
+    @Nullable
+    public List<MinGenInfo> getMinGenInfo() {
+        return snapshot == null ? null : snapshot.minGens();
+    }
+
+    /**
+     * The minimum generation of famous witnesses for the round specified. This method only looks at non-ancient rounds
+     * contained within this state.
+     *
+     * @param round the round whose minimum generation will be returned
+     * @return the minimum generation for the round specified
+     * @throws NoSuchElementException if the generation information for this round is not contained withing this state
+     */
+    public long getMinGen(final long round) {
+        for (final MinGenInfo info : getMinGenInfo()) {
```

Comment:
`getMinGenInfo` might return `null`

Reply:
Pre-existing issue that was copy-pasted into this class. I added the following check, at least the exception will be more meaningful if we ever encounter this in the wild.


```
        final List<MinGenInfo> minGenInfo = getMinGenInfo();
        if (minGenInfo == null) {
            throw new IllegalStateException("No MinGen info found in state for round " + round);
        }

        for (final MinGenInfo info : minGenInfo) {
```